psi4/driver/procrouting/scf_proc/scf_iterator.py

```diff
@@ -267,9 +267,13 @@ def scf_iterate(self, e_conv=None, d_conv=None):
     soscf_enabled = _validate_soscf()
     frac_enabled = _validate_frac()
     efp_enabled = hasattr(self.molecule(), 'EFP')
+    cosx_enabled = "COSX" in core.get_option('SCF', 'SCF_TYPE')
 
     # does the JK algorithm use severe screening approximations for early SCF iterations?
     early_screening = self.jk().get_early_screening()
+    if cosx_enabled:
+        self.jk().set_COSX_grid("Initial")
```

Comment:
shouldn't this be dependent on the use of `early_screening`?

Reply:
What to do with the `early_screening` variable is actually something I wasn't quite sure of with this PR, and I mention this in the Questions section of this PR submission.
 
Disconnecting the `early_screening` variable from control over the COSX grid is an intentional design decision, as there are plans to expand COSX to a three-grid scheme, which is incompatible with `early_screening`'s binary nature as a boolean. I kept it in for now, because "early screening" as a concept can refer to other forms of SCF optimization tricks (e.g., starting early SCF iterations with a looser ERI screening tolerance and tightening the tolerance as the SCF progresses).

But since it's not really being used in this PR, it's probably better to remove it. We wouldn't want dead code, after all.

Reply:
Disconnecting the `early_screening` variable from control over the COSX grid is an intentional design decision in this PR. I plan on eventually adding a three-grid scheme to COSX, where three grids are used over the course of the SCF. This is incompatible with grid control via `early_screening`, given the latter is a boolean.